Buildings/Experimental/DHC/Networks/Steam/BaseClasses/PartialDistributionTwoPipe.mo

```diff
@@ -0,0 +1,197 @@
+    h_outflow(
+      start=MediumRet.h_default,
+      nominal=MediumRet.h_default))
+    "Distribution return outlet port"
+    annotation (Placement(transformation(extent={{-110,-70},{-90,-50}}),iconTransformation(extent={{-220,-80},{-180,-40}})));
+  Modelica.Fluid.Interfaces.FluidPort_a port_aDisRet(
+    redeclare final package Medium=MediumRet,
+    m_flow(
+      min=
+        if allowFlowReversal then
+          -Modelica.Constants.inf
+        else
+          0),
+    h_outflow(
+      start=MediumRet.h_default,
+      nominal=MediumRet.h_default))
+    "Distribution return inlet port"
+    annotation (Placement(transformation(extent={{90,-70},{110,-50}}),iconTransformation(extent={{180,-80},{220,-40}})));
+  // COMPONENTS
+  replaceable Buildings.Experimental.DHC.Networks.Steam.BaseClasses.PartialConnectionTwoPipe con[nCon](
+    final mDis_flow_nominal=mDisCon_flow_nominal,
+    final mCon_flow_nominal=mCon_flow_nominal,
+    each final allowFlowReversal=allowFlowReversal,
+    each final energyDynamics=energyDynamics,
+    each final tau=tau) "Connection to agent"
+    annotation (Placement(transformation(extent={{-10,-10},{10,10}})));
+  Model_pipDis pipEnd(
+    final m_flow_nominal=mEnd_flow_nominal)
+    "Pipe representing the end of the distribution line (after last connection)"
+    annotation (Placement(transformation(extent={{60,-10},{80,10}})));
+initial equation
+  assert(
+    mDis_flow_nominal >= sum(
+      mCon_flow_nominal),
+    "In "+getInstanceName()+": mDis_flow_nominal = "+String(
+      mDis_flow_nominal)+" whereas it must be higher than sum(mCon_flow_nominal) = "+String(
+      sum(
+        mCon_flow_nominal))+".");
+equation
+  // Connecting outlets to inlets for all instances of connection component.
+  if nCon >= 2 then
+    for i in 2:nCon loop
+      connect(con[i-1].port_bDisSup,con[i].port_aDisSup);
+      connect(con[i-1].port_aDisRet,con[i].port_bDisRet);
+    end for;
+  end if;
+
+  connect(con.port_bCon,ports_bCon)
+    annotation (Line(points={{0,10},{0,40},{-80,40},{-80,100}},color={0,0,222},
+      thickness=0.5));
+
+  connect(ports_aCon,con.port_aCon)
+    annotation (Line(points={{80,100},{80,40},{6,40},{6,10}},color={0,0,222},
+      thickness=0.5));
+  connect(port_aDisSup,con[1].port_aDisSup)
+    annotation (Line(points={{-100,0},{-10,0}},color={0,0,222},
+      thickness=0.5));
+  connect(port_bDisRet,con[1].port_bDisRet)
+    annotation (Line(points={{-100,-60},{-40,-60},{-40,-6},{-10,-6}},color={0,0,222},
+      thickness=0.5));
+
+  connect(con[nCon].port_aDisRet,port_aDisRet)
+    annotation (Line(points={{10,-6},{40,-6},{40,-60},{100,-60}},color={0,0,222},
+      thickness=0.5));
+
+  connect(con[nCon].port_bDisSup,pipEnd.port_a)
+    annotation (Line(points={{10,0},{60,0}},color={0,0,222},
+      thickness=0.5));
+  connect(pipEnd.port_b,port_bDisSup)
+    annotation (Line(points={{80,0},{100,0}},color={0,0,222},
+      thickness=0.5));
+
+  annotation (
+    Documentation(
```

Comment:
Both can be correct depending on the circumstance. 

If the same default component name applies to all models that extend that base class (for example, all models that extend the `PartialDistribution` base class are "distribution networks", aka "dis"), then, we should set `defaultComponentName` in the base class. 

If the default name may change across models that extend the base class, then define the default name in the lowest level model that is common to all above that should use that same name. This may be another base class, but at a higher level, or a top-level instantiable model. 

Because `PartialDistributionTwoPipe` extends `PartialDistribution`, it takes on the default name from the inheritance. Thus, we do not need to redeclare it in this model. 

Does that make sense?

It is possible that I did not follow this rule precisely in these models, but for this location, I don't think there are any changes needed. :)